.github/workflows/itest-deploy.yml

```diff
@@ -13,6 +13,62 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
 
+      - name: Install jq
+        run: sudo apt-get install -y jq
+
+      - name: Initialize Comment
+        id: create_comment
+        run: |
+          COMMENT_BODY="{ \"body\": \"| Step | Status |\\n|------|--------|\\n| Initialize Workflow |\" }"
+          COMMENT_RESPONSE=$(curl -s -X POST \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github+json" \
+            -d "$COMMENT_BODY" \
+            https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.issue.number }}/comments)
+          COMMENT_ID=$(echo "$COMMENT_RESPONSE" | jq -r .id)
+          echo "COMMENT_ID=$COMMENT_ID" >> $GITHUB_ENV
+
+      - name: Update Comment Function
+        id: update_comment
+        run: |
+          update_comment() {
+            STEP_NAME="$1"
+            STATUS="$2"
+
+            # Define the icons for each status
+            case "$STATUS" in
+              "Pending") ICON="⏳" ;;
+              "Success") ICON="✅" ;;
+              "Fail") ICON="❌" ;;
+              *) ICON="⏳" ;; # Default to Pending if status is unrecognized
+            esac
+
+            # Fetch the current comment body
+            COMMENT_BODY=$(curl -s -X GET \
+              -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+              -H "Accept: application/vnd.github+json" \
+              https://api.github.com/repos/${{ github.repository }}/issues/comments/${COMMENT_ID} | jq -r .body)
+
+            # Add or update the step's status with the icon
+            if echo "$COMMENT_BODY" | grep -q "| $STEP_NAME |"; then
+              # Update existing step
+              UPDATED_BODY=$(echo "$COMMENT_BODY" | sed "s/| $STEP_NAME |.*|/| $STEP_NAME | $ICON $STATUS |/")
+            else
+              # Append a new step
+              UPDATED_BODY=$(echo -e "$COMMENT_BODY\\n| $STEP_NAME | $ICON $STATUS |")
+            fi
+
+            # Update the comment on GitHub
+            curl -s -X PATCH \
+              -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+              -H "Accept: application/vnd.github+json" \
+              -d "{\"body\": \"$UPDATED_BODY\"}" \
+              https://api.github.com/repos/${{ github.repository }}/issues/comments/${COMMENT_ID} > /dev/null
```

Comment:
Same remark here, please adapt this for pull request comments, rather than issue comments.